### DISCUSSION TITLE
canonical metrics

### DISCUSSION THREAD

**Discussion by mikebell90** [kiali/kiali#6643] [Q&A] [answered]
istio introduced source_canonical_server|namespace and the destination versions.  The rationale is discussed here https://docs.google.com/document/d/1EsKpYNW55-LqG08zdjp1tAd224T9IEG3CEnusmL-Joc/edit

Anyway, we are cleaning up and removing unneeded labels to our metrics. We'd like to remove the canonical versions as for our use cases we don't need them - we just use the noncanonical versions.

Kiali's faq https://kiali.io/docs/faq/general/#requiredmetrics claims source_canoncial_* is used. Confusingly destination_canonical is not listed

So the questions at last

1. Does kiali indeed use these "canonical" labels? What for? 
2. Can we remove them

**Comment by jshaughn** [comment, marked as answer]
Hi, you can't remove them if you want Kiali to function properly.  I'd actually suggest you use the `canonical` fields and not the deprecated attributes like 'app' and 'version'.  Looking at https://kiali.io/docs/faq/general/#requiredmetrics, I see listed both the destination and source variations of the canonical attributes...

We use those attributes in many places, for example, to generate the graph.

**Reply by ArnoldZokas** [reply]
> I'd actually suggest you use the canonical fields and not the deprecated attributes like 'app' and 'version'

Are you specifically referring to `source_app`, `source_version`,`destination_app`, `destination_version`?
When were these deprecated?
Do you have a link to the announcement?

**Reply by jshaughn** [reply]
I can't find an announcement, I must have been thinking about discussions about possible deprecation given that the canonical attributes are preferred.  So, strike my comment about deprecation, but not my recommendation to use the canonical fields instead.

**Reply by ArnoldZokas** [reply]
Thanks for the clarification.
